### DISCUSSION TITLE
Variable naming for identical variables: Global vs Local

### DISCUSSION THREAD

**Discussion by PeterConijn** [microsoft/alguidelines#236] [BC Best Practices]
In our organization, questions have been raised on how we handle global and local variable names for identical variables. The reason this came up is because AppSourceCop is warning about it (AA0244).

So the question became: what is the preferred way of solving this? I have seen that Microsoft tends to affix some global variables with Global (e.g. SalesLineGlobal) and sometimes they affix parameters with "New" (e.g. NewLocationCode). 

Since AppSourceCop requires affixes which can be either prefixes or suffixes, having a universal preferred method is tricky. Do we go SalesLineGlobalXXX or GlobalSalesLineXXX? NewLocationCodeXXX or LocationCodeNewXXX?

Similar questions are raised for situations of DataItem vs. Variable.

Thoughts?

**Comment by PeterConijn** [comment]
Ps. once (if) we reach a consensus, I will create an issue and PR and add it to the page.

**Comment by NKarolak** [comment]
[AA0244](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0244) is about parameters vs globals. Is your discussion about naming all non-global variables, or the parameters specifically?

Speaking generally, I tend to update the local names instead of the global names, but there are different scenarios. In one scenario, I tend to prefix the parameter with `New*` Only if there are many local duplicates, I tend to prefix the global with `Global*`.

**Reply by fvet** [reply]
We apply the same rules internally. **New** and **Global** covers most of the scenarios ...

Extract from our guidelines...

![image](https://github.com/microsoft/alguidelines/assets/25268332/b8c93e79-7154-4fab-93c8-523df5e77217)
